lib/_http_server.js

```diff
@@ -603,6 +603,7 @@ function resOnFinish(req, res, socket, state, server) {
   assert(state.incoming.length === 0 || state.incoming[0] === req);
 
   state.incoming.shift();
+  if (socket.parser) socket.parser.incoming = null;
```

Comment:
Is it worth adding a comment explaining why this is being done? Or is setting it to `null` idiomatic for memory management?

Reply:
@trott I think it’s idiomatic, but I’m happy to add something like 

```suggestion
  // Reset the .incoming property so that the request object can be gc'ed.
  if (socket.parser) socket.parser.incoming = null;
```

Reply:
Eh, if it's idiomatic, let's leave it.

Reply:
Or not. Looks like you added the comment, which is totally fine too. :-D